### PR TITLE
Adds multiple character selection

### PR DESCRIPTION
**Pull request by MaxIsJoe**
### Purpose
Ticks one box from #6276 
Adds multiple characters support and the ability to select between them.

### Notes:

After a bit of struggling with unity's cameras and loading images, It's done! Players now can create an unlimited amount of characters to choose from to roleplay as!

How does this work? All characters are currently stored in a JSON file with all their data and customizations, when the character page opens it loads that JSON file and adds all characters to a list that the player can browse in a new (functional but not all that pretty) UI. Just land on the character you want to use then that's your character that you'll be using for this round.
Don't have a character? Want to edit your current character? no worries, you can do that from the two buttons at the bottom of the the character selector page.

This PR doesn't just come with a character selector, I've also added in two scripts at the request of Bod to capture pictures of the game/UI. This can be used later when we work on fixing the security console and add in some functionality to cameras so we can say cheese. 

PS : I've been told that the UI/UX peeps will handle the looks of this later so I've prioritized functionality over shininess.

### Changelog:
CL: Multiple character sheets are now supported.
CL: Humans will no longer have wrong skin tones when switching between species.

### CODE REVIEW

**Comment by PerfectTangent** [comment]
when you want to make sure maintainers can't merge a PR that isn't quite finished, you can instead open it as a draft PR.

**Comment by MaxIsJoe** [comment]
> when you want to make sure maintainers can't merge a PR that isn't quite finished, you can instead open it as a draft PR.

Will keep in mind that for next time

**Comment by PerfectTangent** [comment]
for your transgressions, I will now speedmerge this PR /s

**Comment by MaxIsJoe** [comment]
> for your transgressions, I will now speedmerge this PR /s

image

**Comment by MrAsbestos** [comment]
Hey btw you can actually convert it into draft

Look around here
image

**Comment by MaxIsJoe** [comment]
For some reason the camera refuses to render anything which as a result ends up outputting empty pictures.
anyone knows why this is happening?

**Comment by MaxIsJoe** [comment]
Why is this tagged with "Animation" ?

**Comment by PerfectTangent** [comment]
because working with cameras and capturing static sprite compositions is animation as far as unity is concerned. Read the description of the label to see the general scope that it covers.

**Comment by MaxIsJoe** [comment]
review

**Review by MaxIsJoe — COMMENTED** [review]
review time

**Comment by PerfectTangent** [comment]
@Bod9001 can you help us diagnose some of the bugs we've been experiencing with the character customizer and saving characters?

**Comment by Bod9001** [comment]
oh? what would you like me to work on

**Comment by MaxIsJoe** [comment]
> oh? what would you like me to work on

Characters are being saved incorrectly, where for some reason the game will save two indexes instead of one for no reason even though I have explicitly told it multiple times to save the current character I'm creating/editing instead of the older ones.

**Comment by MaxIsJoe** [comment]
Requesting a new review and testing for all the new fixes. Bod's help may no longer be needed.

Edit, not sure what's with codacy and async. Anyone can explain why it wants me to make that change?

File: UnityProject/Assets/Scripts/UI/Systems/Lobby/CharacterCustomization.cs [hunk 1]
```diff
@@ -745,10 +908,89 @@ private void SaveData()
...
+			string path = Application.persistentDataPath + "characters.json";
+			if(File.Exists(path))
+			{
+				File.Delete(path);
+			}
+			File.WriteAllText(path, json);
+		}
+
+		/// <summary>
+		/// Get all characters that are saved in %APPDATA%/Locallow/unitystation/characters.json
```
Comment (Bod9001) [inline UnityProject/Assets/Scripts/UI/Systems/Lobby/CharacterCustomization.cs:974]: assume this changes per operating system

Comment (MaxIsJoe) [reply UnityProject/Assets/Scripts/UI/Systems/Lobby/CharacterCustomization.cs:974]: https://docs.unity3d.com/ScriptReference/Application-persistentDataPath.html

**Comment by MaxIsJoe** [comment]
Anything else needs to be checked?

**Comment by PerfectTangent** [comment]
I'd like to give it another go in the editor when I wake up

**Comment by Bod9001** [comment]
oh yeah does this have the box for serialising and de-serialising text you input for the character selected?

**Comment by MaxIsJoe** [comment]
> oh yeah does this have the box for serialising and de-serialising text you input for the character selected?

Yep, just edit or create a new character and give it a character or get your character's json thingy.

**Comment by Bod9001** [comment]
ok cool, can merge after resolving conflict

**Comment by MaxIsJoe** [comment]
> ok cool, can merge after resolving conflict

There are a few minor issues listed by @PerfectTangent , these issues may or may have not been fixed so until we've made sure that characters no longer spawn with wrong character sheets we shouldn't merge this.
Note : Issues like "Lizards not having their tails colored" or #6323 are not covered under this PR. This is mainly to add the ability to switch between characters, not introduce any other fixes or changes as they're going to be covered under different PRs for #6276 .

Someone test this PR a couple more times and report any issues you find related to character switching.

**Comment by PerfectTangent** [comment]
agreed. tagged appropriately

**Comment by MaxIsJoe** [comment]
I've fixed some more issues for the character creator, ready for one more review.
There should be no longer any issues related to characters spawning with wrong sheets, editing the wrong character sheets, buttons not being clickable and layering issues when capturing previews.

**Comment by MaxIsJoe** [comment]
Everything seems to be fine from testing, no more issues related to this PR.

We can merge now.

**Comment by MaxIsJoe** [comment]
Whoops, I forgot to push this change 2 days ago.
Codacy can now shut the fuck up about this task/void error.

Reminder again, we're done with this PR. We're ready to merge.

**Comment by MaxIsJoe** [comment]
ok for real this time, no more issues.

**Pull request merged**